plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-shipping-methods-block/block.tsx

```diff
@@ -73,19 +74,22 @@ const Block = ( { noShippingPlaceholder = null } ): ReactElement | null => {
 
 	const { shippingAddress } = useCustomerData();
 
-	const filteredShippingRates = isCollectable
-		? shippingRates.map( ( shippingRatesPackage ) => {
-				return {
-					...shippingRatesPackage,
-					shipping_rates: shippingRatesPackage.shipping_rates.filter(
-						( shippingRatesPackageRate ) =>
-							! hasCollectableRate(
-								shippingRatesPackageRate.method_id
-							)
-					),
-				};
-		  } )
-		: shippingRates;
+	const memoizedFilteredShippingRates = useMemo( () => {
```

Comment:
nitpick in my own code that this variable should just be named `filteredShippingRates`